pkg/noderesourcetopology/cache/overallocating.go

```diff
@@ -0,0 +1,238 @@
+// This function enables the caller to know the slice of nodes should be considered for resync,
+// avoiding the need to rescan the full node list.
+func (ov *Overallocating) DirtyNodeNames(logKey string) []string {
+	ov.lock.Lock()
+	defer ov.lock.Unlock()
+	var nodes []string
+	for node := range ov.assumedResources {
+		if !ov.nodeDiscarded.IsSet(node) {
+			// noone asked about this node, let it be
+			continue
+		}
+		nodes = append(nodes, node)
+	}
+
+	if len(nodes) > 0 {
+		klog.V(4).InfoS("nrtcache: found dirty nodes", "logKey", logKey, "nodes", len(nodes))
+	}
+	return nodes
+}
+
+// Resync implements the cache resync loop step. This function checks if the latest available NRT information received matches the
+// state of a dirty node, for all the dirty nodes. If this is the case, the cache of a node can be Flush()ed.
+// The trigger for attempting to resync a node is not just that we overallocated it. If a node was overallocated but still has capacity,
+// we keep using it. But we cannot predict when the capacity is too low, because that would mean predicting the future workload requests.
+// The best heuristic found so far is count how many times the node was skipped *AND* crosscheck with its overallocation state.
+// If *both* a node has pessimistic overallocation accounted to it *and* was discarded "too many" (how much is too much is a runtime parameter
+// which needs to be set and tuned) times, then it becomes a candidate for resync. Just using one of these two factors would lead to
+// too aggressive resync attempts, so to more, likely unnecessary, computation work on the scheduler side.
+func (ov *Overallocating) Resync() {
+	// we are not working with a specific pod, so we need a unique key to track this flow
+	logKey := logKeyFromTime()
+	nodeNames := ov.DirtyNodeNames(logKey)
+	// avoid as much as we can unnecessary work and logs.
+	if len(nodeNames) == 0 {
+		return
+	}
+
+	klog.V(6).InfoS("nrtcache: resync NodeTopology cache starting", "logKey", logKey)
+	defer klog.V(6).InfoS("nrtcache: resync NodeTopology cache complete", "logKey", logKey)
+
+	var nrtUpdates []*topologyv1alpha1.NodeResourceTopology
+	for _, nodeName := range nodeNames {
+		nrtCandidate, err := ov.lister.Get(nodeName)
+		if err != nil {
+			klog.V(3).InfoS("nrtcache: failed to get NodeTopology", "logKey", logKey, "node", nodeName, "error", err)
+			continue
+		}
+		if nrtCandidate == nil {
+			klog.V(3).InfoS("nrtcache: missing NodeTopology", "logKey", logKey, "node", nodeName)
+			continue
+		}
+
+		pfpExpected := podFingerprintForNodeTopology(nrtCandidate)
+		if pfpExpected == "" {
+			klog.V(3).InfoS("nrtcache: missing NodeTopology podset fingerprint data", "logKey", logKey, "node", nodeName)
+			continue
+		}
+
+		klog.V(6).InfoS("nrtcache: trying to resync NodeTopology", "logKey", logKey, "node", nodeName, "fingerprint", pfpExpected)
+
+		err = checkPodFingerprintForNode(logKey, ov.indexer, nodeName, pfpExpected)
```

Comment:
I guess this is always consistent as ov.indexer is always consistent with ov.lister, and hence consistent with pfpExpected. We should compare the reserved pods (instead of pods in ov.indexer) with pfpExpected?

Reply:
Uhm, if we stop caching NRT data (per suggestion in https://github.com/kubernetes-sigs/scheduler-plugins/pull/437#discussion_r1021394760) we can indeed compare the pods alone